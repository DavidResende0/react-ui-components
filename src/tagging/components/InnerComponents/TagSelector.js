import React from 'react';
import PropTypes from 'prop-types';
import { __ } from '../../../global-functions';
import TaggingPropTypes from '../TaggingPropTypes';
import Select from '../../../forms/pf-select';

class TagSelector extends React.Component {
  handleChange = (selectedOption) => {
    this.props.onTagCategoryChange({
      id: selectedOption.value,
      description: this.props.tagCategories.find(category => category.id === selectedOption.value).description,
    });
  };


  labelWithIcon = (description, infoText) => (
    <div>
      <span
        style={{
          display: 'inline-block',
          width: 'calc(100% - 18px)',
        }}
      >
        {description}
      </span>
      <span
        className="pull-right pficon pficon-info tag-icon"
        title={infoText}
        aria-hidden="true"
      />
      <span className="sr-only">{infoText}</span>
    </div>
  );

  tagCategories = this.props.tagCategories.map(category => ({
    keyWord: category.description.toLowerCase(),
    value: category.id,
    label: category.singleValue
      ? this.labelWithIcon(category.description, this.props.infoText)
      : category.description,
  }));


  render() {
    const value = this.props.selectedOption.id ? { label: this.props.selectedOption.description, value: this.props.selectedOption.id } : null;
    return (
      <Select
        meta={{}}
        options={this.tagCategories}
        input={{ onChange: this.handleChange, name: 'form-field-name', value }}
        filterOption={(option, filter) =>
          option.data.keyWord.includes(filter.toLowerCase())
        }
        placeholder={__('Select tag category')}
        id="tag_cat"
        // clearable={false}
        simpleValue={false}
      />
    );
  }
}
TagSelector.propTypes = {
  tagCategories: PropTypes.arrayOf(TaggingPropTypes.category).isRequired,
  selectedOption: TaggingPropTypes.value,
  onTagCategoryChange: PropTypes.func.isRequired,
  infoText: PropTypes.string,
};

TagSelector.defaultProps = {
  infoText: __('Only a single value can be assigned from these categories'),
  selectedOption: {},
};

export default TagSelector;
