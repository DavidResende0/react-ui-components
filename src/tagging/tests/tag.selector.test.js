import React from 'react';
import TagSelector from '../components/InnerComponents/TagSelector';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const tagCategories = [
  { description: 'Name', id: 1 },
  { description: 'Number', id: 2 }
];
const selectedTagValue = { description: 'Homer', id: 1 };
const onTagCategoryChange = jest.fn();

describe('Tagging modifier', () => {
  const tagCategories = [
    { description: 'Name', id: 1 },
    { description: 'Number', id: 2 },
  ];
  const selectedTagValue = { description: 'Homer', id: 1 };
  let onTagCategoryChange;
  beforeEach(() => {
    onTagCategoryChange = jest.fn();
  });

  it('match snapshot', () => {
<<<<<<< HEAD
    const tree = mount(<TagSelector
      tagCategories={tagCategories}
      onTagCategoryChange={onTagCategoryChange}
      selectedOption={selectedTagValue}
    />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should call methods', () => {
    onTagCategoryChange = jest.fn();
    const wrapper = shallow(<TagSelector
      tagCategories={tagCategories}
      onTagCategoryChange={onTagCategoryChange}
      selectedOption={selectedTagValue}
    />);
=======
    const component = renderer.create(
      <TagSelector
        tagCategories={tagCategories}
        onTagCategoryChange={onTagCategoryChange}
        selectedOption={selectedTagValue}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call methods', () => {
    const wrapper = shallow(
      <TagSelector
        tagCategories={tagCategories}
        onTagCategoryChange={onTagCategoryChange}
        selectedOption={selectedTagValue}
      />
    );
>>>>>>> Change tests
    wrapper.instance().handleChange({ label: 'Name', value: 1 });
    expect(onTagCategoryChange.mock.calls).toHaveLength(1);
  });
});
