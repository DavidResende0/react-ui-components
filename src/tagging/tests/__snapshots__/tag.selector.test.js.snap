// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tagging modifier match snapshot 1`] = `
<StateManager
  className="final-form-select"
  clearable={false}
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  filterOptions={[Function]}
  name="form-field-name"
  onChange={[Function]}
  optionClassName="selected-option final-form-select-option"
  options={
    Array [
      Object {
        "keyWord": "name",
        "label": "Name",
        "value": 1,
      },
      Object {
        "keyWord": "number",
        "label": "Number",
        "value": 2,
      },
    ]
  }
  placeholder="Select tag category"
  styles={
    Object {
      "clearIndicator": [Function],
      "container": [Function],
      "control": [Function],
      "dropdownIndicator": [Function],
      "indicatorSeparator": [Function],
      "indicatorsContainer": [Function],
      "input": [Function],
      "menu": [Function],
      "multiValue": [Function],
      "multiValueLabel": [Function],
      "multiValueRemove": [Function],
      "option": [Function],
      "placeholder": [Function],
      "singleValue": [Function],
      "valueContainer": [Function],
    }
  }
  value={
    Object {
      "label": "Homer",
      "value": 1,
    }
  }
/>
`;
